src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -526,7 +553,6 @@ AllSyntaxTreesParsedState allSyntaxTreesParsedState
                             => await FinalizeCompilationAsync(
                                 compilationState,
                                 allSyntaxTreesParsedState,
-                                compilationWithStaleGeneratedTrees: null,
```

Comment:
note: @jasonmalinowski this is a change in behavior in that *now* we will use the stale-compilation off of allSyntaxTreesParsedState (which might not be null afaict).  However, i think this is safe as i *think* the stale-compilation just acts as something we can reuse if *everything* about it matches up.  But i'm not 100% sure.  i need your eyes on this part of things and how you think that data flows forward and is utilized.

Reply:
That seems entirely reasonable -- did we regress this recently? I imagine the only way we might have hit this before would have been some sequence of:

1. Ask for a compilation.
2. We get all trees up to date, and then we set it as an AllSyntaxTreesParsed state.
3. We get lucky and cancel during finalization.
4. Somebody else asks again, so we're now using the state from 2.

At which point absolutely we should be using it.

Reply:
great!  glad we think this is good.